### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by davidkskim**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -4,32 +4,62 @@
...
+        return `${temp} is above boiling point`
+    }
+    //checks for temp that is boiling point.
+    else {
+        return `${temp} is at boiling point`
+    }
+}
+console.log(tempPoint(temp1));
+console.log(tempPoint(temp2));
+console.log(tempPoint(temp3));
```
Comment (andeeisaacs) [inline code-challenges.js:30]: Excellent!

File: code-challenges.js [hunk 2]
```diff
@@ -4,32 +4,62 @@
...
 var myNumbers1 = [3, 7, 0, 6, -9]
-
-
+const multiply = (array) => {
+    //create new array to put modified #'s in.
+    let newArray = []
+    //loop through each index.
+    for (let i = 0; i < array.length; i++) {
+        //push the index value multiplied by 5 into newArray.
+        newArray.push(array[i] * 5)
```
Comment (andeeisaacs) [inline code-challenges.js:44]: Excellent use of push here.

File: code-challenges.js [hunk 3]
```diff
@@ -4,32 +4,62 @@
...
 // Use the test variable provided below. Expected outcome: [40, -35, 0, 30, 10]
 
 var myNumbers2 = [8, -7, 0, 6, 2]
-
-
+const multiplier = (array) => {
+    //gives back index value that is multiplied by 5. 
+    return array.map(value => {
+        return value * 5
+    })
```
Comment (andeeisaacs) [inline code-challenges.js:60]: Awesome! Another way we can do this, since we're only checking for one thing, is to not open up our code block. 

`return array.map(value => value * 5)`
This is doing the EXACT same thing, but just a little prettier. 

It's totally fine how you have it already, just wanted to show you another way :)

File: code-challenges.js [hunk 4]
```diff
@@ -38,8 +68,14 @@ var myNumbers2 = [8, -7, 0, 6, 2]
...
-
-
+const vowelRem = (string) => {
+    //separeates string into array, filters out any vowels and puts them into a new array, then joins the array together to form a string again.
+    return string.split("").filter(value => {
+        return value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u"
+    }).join("")
+}
+console.log(vowelRem(stringWithVowels1));
+console.log(vowelRem(stringWithVowels2));
```
Comment (andeeisaacs) [inline code-challenges.js:78]: Oh! Really close! 

Let's take a look here at your function output versus the expected output- 

First, your output is `ILvJvScrpt`'

But our expected output is `LvJvScrpt`

This means, it's not catching the capitalized 'I' vowel. The way we can fix this is to turn our values into lower case letters, I think .toLowerCase() would catch this on each value check!

File: code-challenges.js [hunk 5]
```diff
@@ -48,19 +84,54 @@ var stringWithVowels2 = "ILoveJavaScript"
...
+        return `${string} is not a string`
+    }
+    else {
+        return string.split("").filter(value => {
+            return value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u"
+        }).join("")
+    }
+}
+console.log(vowelRem2(notAString1));
+console.log(vowelRem2(notAString2));
```
Comment (andeeisaacs) [inline code-challenges.js:99]: Great!

File: code-challenges.js [hunk 6]
```diff
@@ -48,19 +84,54 @@ var stringWithVowels2 = "ILoveJavaScript"
...
+var toonimals = [{ name: "Itchy", animal: "mouse" },
+{ name: "Stimpy", animal: "cat" },
+{ name: "Daffy", animal: "duck" },
+{ name: "Scratchy", animal: "cat" },
+{ name: "Ren", animal: "dog" },
+{ name: "Felix", animal: "cat" }]
+const toonObjects = (array) => {
+    //filters out objects that are not cats.
+    return array.filter(value => {
+        return value.animal === "cat"
```
Comment (andeeisaacs) [inline code-challenges.js:115]: Similar to when we were mapping through the array previously, since we're only checking for one thing, its not necessary to open the code block. 

In this case, it can be a little more tidy- `return array.filter(value => value.animal === "cat" `

Again, nothing wrong with how you have it, just something to keep in mind if we are only checking for one condition.

File: code-challenges.js [hunk 7]
```diff
@@ -48,19 +84,54 @@ var stringWithVowels2 = "ILoveJavaScript"
...
+    let animal = array.filter(value => {
+        return value.animal !== "cat"
+    })
+    //returns a new array that holds the names of the animals.
+    let names = animal.map(value => {
+        return value.name
+    })
+    //joins the names of animals from array to string.
+    return names.join(" ")
+}
```
Comment (andeeisaacs) [inline code-challenges.js:136]: This is GREAT it works super well! 

I want to show you another way we can do this- all in one single line! 
`const noCats = (array) => {`
   ` return array.filter(value => value.animal !== "cat").map(value => value.name).join(" ")`
`}`

So whats going on up in this example? Well, we return the array immediately, and we say to the return, we want the array filtered to only show the cats out of the array, then we chained on the map to also only give us the names. We then of course then chain on the .join. 

Its doing the exact same thing you are, but all tied up nicely together without having to create extra variables! Just something to think about!

File: interview-questions.md [hunk 8]
```diff
@@ -7,64 +7,82 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: Functions are ways to store directions all onto one word by taking in an argument.
```
Comment (andeeisaacs) [inline interview-questions.md:10]: Just about- Its also about having reusable blocks of code that allows you to also take in different sets of data and manipulate it. You covered this well with your researched answer!

File: interview-questions.md [hunk 9]
```diff
@@ -7,64 +7,82 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  Researched answer: A function is a block of organized, reusable code that is used to perform a single, related action. 
+  
+  (https://www.tutorialspoint.com/computer_programming/computer_programming_functions.htm#:~:text=Advertisements,high%20degree%20of%20code%20reusing.)
 
 
 
 2. What is the difference between .map() and .filter()?
 
-  Your answer:
+  Your answer: .map() allows users to reformat all index values of an array, whereas .filter() takes out certain index values.
```
Comment (andeeisaacs) [inline interview-questions.md:20]: Pretty much, but I think it can be expanded upon much further. Your researched answer gets into it a bit better :)

File: interview-questions.md [hunk 10]
```diff
@@ -7,64 +7,82 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  Researched answer: They do different things: . filter() returns a subset of the elements from the original array, while . map() produces an array with new, different entries based on the elements in the original array. 
+  
+  (https://discuss.codecademy.com/t/what-s-the-difference-between-filter-and-map/342066)
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log() prints out the results of the called function on the terminal, whereas return gives you back a value that you are specifically looking for and is stored in the function.
```
Comment (andeeisaacs) [inline interview-questions.md:30]: Console.log actually can be used in many places! Not just in a function! Its also mostly used for behind the scenes work, such as debugging since it will only be seen by a terminal or by a console in your browser. 

Additionally a return is what stops your function and provides an output from it when its called.

File: interview-questions.md [hunk 11]
```diff
@@ -7,64 +7,82 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  Researched answer: Return is used to interrupt the flow of execution of a function and return some value (or not). console.log() is used to print information to the console (acessible by F12) of the browser. 
+  
+  (https://www.quora.com/What-is-the-difference-between-console-log-and-return-in-JavaScript#:~:text=Difference%20between%20console.-,log%20and%20return%20in%20javascript%3F,by%20F12)%20of%20the%20browser.)
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument is what you run through the function to reformat.
```
Comment (andeeisaacs) [inline interview-questions.md:40]: Interesting way of describing it! 🤔

File: interview-questions.md [hunk 12]
```diff
@@ -7,64 +7,82 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+(https://www.quora.com/What-does-an-Argument-in-Programming-mean)
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Give each person an equal amount of work to do. Communicate communicate communicate. Don't be afraid of each other.
+               Navigator - the one who reads the instructions and tells the driver what to write down.
+               Driver - the one who listens to the navigator and writes the code down.
```
Comment (andeeisaacs) [inline interview-questions.md:52]: Totally!

File: interview-questions.md [hunk 13]
```diff
@@ -7,64 +7,82 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  (https://en.wikipedia.org/wiki/Pair_programming#:~:text=Pair%20programming%20is%20an%20agile,two%20programmers%20switch%20roles%20frequently.)
 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer: I don't remember what TDD is or stands for...
```
Comment (andeeisaacs) [inline interview-questions.md:62]: Thats ok! 

You'll hear this later on when dealing with interview questions, but if you dont know the answer, honesty is definitely the best policy, but there is a way to back it up and come out on top- "I'm not sure, but I know I can look it up at .. and .. " replace the blank spots where you would say perhaps your notes, or your syllabus, or official documentation or what you would google! 

This would show a proactiveness, even if you don't know the answer right off the bat, to finding the answer and that is super valuable to a almost every company.

File: testing-with-jest.test.js [hunk 14]
```diff
@@ -7,21 +7,31 @@
 
 
 // --------------------1a) Create a test for a function called addThemUp that takes two numbers as arguments and returns the sum.
-
-
+describe("addThemUp", () => {
+    test("Takes two numbers as arguments and returns the sum.", () => {
+        expect(addThemUp(4, 4)).toEqual(16)
+    })
+})
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:14]: Excellent!

File: testing-with-jest.test.js [hunk 15]
```diff
@@ -7,21 +7,31 @@
...
 
 
 // --------------------2a) Create a test for a function called triangleArea that takes the base and the height of a triange and returns the area. The area of a triange is the base times the height divided by two.
-
-
+describe("triangleArea", () => {
+    test("takes the base and the height of a triange and returns the area.", () => {
+        expect(triangleArea(4, 10)).toEqual(20)
+    })
+})
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:30]: Fantastic!

**Review by andeeisaacs — APPROVED** [review]
David you did great! Your solutions were very good and you managed to pseudo code almost everything!

Just a few pieces of advice:
- Pseudo code more! I want to see more of how you are figuring out what built in functions you used and the flow of your code. Keep it up! 
- Your tests are good! But I would like to see more than a single expect, check for more variations! Check for some edge cases! What if they put in a datatype that is unexpected? 
- Double-tripple check your expected outcomes versus what your functions are outputting. Two of of your functions are not currently checking for the necessary edge cases, thus your outputs are incorrect. 

Overall, you did FANTASTIC for your first week! You should feel proud. 

Feel free to merge!
